src/interface/src/app/services/plan.service.ts

```diff
@@ -338,6 +358,42 @@ export class PlanService {
     this.planState$.next(updatedState);
   }
 
+  updateStateWithConfig(configId: number | null) {
+    const currentState = Object.freeze(this.planState$.value);
+    const updatedState = Object.freeze({
+      ...currentState,
+      all: {
+        ...currentState.all,
+      },
+      currentConfigId: configId,
+    });
+    this.planState$.next(updatedState);
+  }
+
+  updateStateWithConditionFilepath(filepath: string | null) {
```

Comment:
nice! thanks for refactoring! this is what I had in mind too to display the project areas in the plan map in scenario details view.